tests/test_main.py

```diff
@@ -1,11 +1,125 @@
+import uuid
+import pytest
 from fastapi.testclient import TestClient
-
 from src.main import app
+from typing import Generator
+
+from src.schemas import Project, ProjectDetails
+
+
+@pytest.fixture
+def client() -> Generator[TestClient, None, None]:
+    yield TestClient(app)
+
+
+@pytest.fixture(autouse=True)
+def clear_projects() -> Generator[None, None, None]:
+    from src.main import projects
+
+    projects.clear()
+    yield
+    projects.clear()
+
+
+@pytest.fixture
+def sample_project() -> Project:
+    return Project(name="Project 1", description="Test project")
+
+
+@pytest.fixture
+def sample_project_2() -> Project:
+    return Project(name="Project 2", description="Another test project")
+
+
+@pytest.fixture
+def created_project(sample_project: Project) -> ProjectDetails:
+    from src.main import projects
+
+    project_details = ProjectDetails(**sample_project.model_dump())
+    projects.append(project_details)
+    return project_details
+
+
+@pytest.fixture
+def created_projects(client: TestClient, sample_project: Project, sample_project_2: Project) -> list[ProjectDetails]:
+    projects = []
+    for project in [sample_project, sample_project_2]:
+        response = client.post("/projects", json=project.model_dump())
+        projects.append(ProjectDetails(**response.json()))
+    return projects
 
-client = TestClient(app)
 
+def test_create_project(client: TestClient, sample_project: Project) -> None:
+    response = client.post("/projects", json=sample_project.model_dump())
+    assert response.status_code == 201
+    data = response.json()
+    assert data["name"] == sample_project.name
+    assert data["description"] == sample_project.description
+    assert "project_id" in data
 
-def test_read_root() -> None:
-    response = client.get("/")
+
+def test_get_projects(client: TestClient, created_projects: list[ProjectDetails]) -> None:
+    response = client.get("/projects")
+    assert response.status_code == 200
+    data = response.json()
+    assert all("project_id" in project for project in data)
+    assert data[0]["name"] == created_projects[0].name
+    assert data[1]["name"] == created_projects[1].name
+    assert data[0]["description"] == created_projects[0].description
+    assert data[1]["description"] == created_projects[1].description
+
+
+def test_get_project_details(client: TestClient, created_project: ProjectDetails) -> None:
+    response = client.get(f"/project/{created_project.project_id}/info")
+    assert response.status_code == 200
+    data = response.json()
+    assert data["name"] == created_project.name
+    assert data["description"] == created_project.description
+    assert data["project_id"] == str(created_project.project_id)
```

Comment:
Same as previous comments - compare the whole output instead of picking up specific fields